src/components/auth/LinkedInLogin.vue

```diff
@@ -0,0 +1,59 @@
+<script setup lang="ts">
+import { onMounted } from 'vue'
+
+import { useRoute } from 'vue-router'
+
+import useAuthStore from '@/stores/auth.store'
+
+import { linkedInScopes } from '@/config/components/auth/linkedInPermissions.config'
+
+const route = useRoute()
+
+const props = defineProps({
+  brandId: {
+    type: String,
+    required: true
+  }
+})
+
+const authStore = useAuthStore()
+
+const linkedInClientId = import.meta.env.VITE_LINKEDIN_CLIENT_ID
+const redirectUri =
+  'https://d811-2800-bf0-8143-f0-c5f4-6c27-3530-85de.ngrok-free.app'
+
+function loginWithLinkedIn() {
+  const state = Math.random().toString(36).substring(2)
```

Comment:
I suggest we have to validate this state to avoid attacks CSRF


```suggestion
  const state = Math.random().toString(36).substring(2)
  sessionStorage.setItem('oauthStaet', state)
```